### PR TITLE
Supported TSQLs information_schema_tsql.key_column_usage view

### PR DESCRIPTION
**Pull request by siddhijain2**
We have supported TSQLs information_schema_tsql.key_column_usage view. Returns one row for each column that is constrained as a key in the current database. This information schema view returns information about the objects to which the current user has permissions.

When tested with Database containing the following no. of rows in respective Catalogs -
pg_attribute - 185380
pg_class - 40595
pg_constraint - 40225
sys.babelfish_namespace_ext - 10008
sys.pg_namespace_ext - 10008

For query -
`SELECT count(*) FROM information_schema.key_column_usage`

Query Plan - https://explain.dalibo.com/plan/3af6a3363e095d9g
Execution time: 266ms
Planning time: 11.8ms

Network packet size (bytes): 4096
1 xact[s]:
Clock Time (ms.): total 305 avg 305.0 (3.3 xacts per sec.)

Received the following performance statistics from performance framework -

```
Query - "SELECT * FROM information_schema.key_column_usage;"
Average Time - 576
Run Count - 1
Runtime - 576 
SLA - 3000
Standard Deviation - 0.000000
```
Query Plan generated by performance framework -
https://explain.dalibo.com/plan/172h213c98fc3d9a

The Baseline Database of performance framework contains the following number of rows in the respective catalogs -

pg_attribute - 88968
pg_class - 12772
pg_constraint - 7472
sys.babelfish_namespace_ext - 8
sys.pg_namespace_ext - 8

### Issues Resolved

[BABEL-3884
](https://jira.rds.a2z.com/browse/BABEL-3884)
### Test Scenarios Covered ###
* **Use case based -** Yes

* **Boundary conditions -** N/A

* **Arbitrary inputs -** N/A

* **Negative test cases -** N/A

* **Minor version upgrade tests -** N/A

* **Major version upgrade tests -** N/A

* **Performance tests -** Added in Performance Framework

* **Tooling impact -** N/A

* **Client tests -** N/A

### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by shah-nirmit — APPROVED** [review]

File: contrib/babelfishpg_tsql/sql/information_schema_tsql.sql [hunk 1]
```diff
@@ -752,6 +752,31 @@ CREATE OR REPLACE VIEW information_schema_tsql.SEQUENCES AS
...
+		CAST(c.conname AS sys.nvarchar(128)) AS "CONSTRAINT_NAME",
+		CAST(nr.dbname AS sys.nvarchar(128)) AS "TABLE_CATALOG",
+		CAST(ext.orig_name AS sys.nvarchar(128)) AS "TABLE_SCHEMA",
+		CAST(r.relname AS sys.nvarchar(128)) AS "TABLE_NAME",
+		CAST(a.attname AS sys.nvarchar(128)) AS "COLUMN_NAME",
+		CAST(ord AS int) AS "ORDINAL_POSITION"	
+	FROM
+		pg_constraint c 
+		JOIN sys.pg_namespace_ext nc ON (nc.oid = c.connamespace)
+		JOIN pg_class r ON r.oid = c.conrelid AND c.contype in ('p','u','f')
```
Comment (Deepesh125) [inline contrib/babelfishpg_tsql/sql/information_schema_tsql.sql]: add filter r.relkind in ('r','p') here

File: contrib/babelfishpg_tsql/sql/information_schema_tsql.sql [hunk 2]
```diff
@@ -752,6 +752,31 @@ CREATE OR REPLACE VIEW information_schema_tsql.SEQUENCES AS
...
+		JOIN sys.pg_namespace_ext nr ON r.relnamespace = nr.oid AND r.relkind in ('r','p')
```
Comment (Deepesh125) [inline contrib/babelfishpg_tsql/sql/information_schema_tsql.sql]: isnt this join redundant?

File: test/JDBC/input/key_column_usage-vu-prepare.sql [hunk 3]
```diff
@@ -0,0 +1,29 @@
+CREATE DATABASE key_column_usage_vu_prepare_db;
+GO
+
+USE key_column_usage_vu_prepare_db;
+GO
+
+CREATE TABLE key_column_usage_vu_prepare_tbl1(arg1 int, arg2 int, primary key(arg1));
```
Comment (Deepesh125) [inline test/JDBC/input/key_column_usage-vu-prepare.sql:7]: test cases with check constraints and unique constraints should be added

**Review by Deepesh125 — CHANGES_REQUESTED** [review]

File: contrib/babelfishpg_tsql/sql/information_schema_tsql.sql (hunk 1 above)
Comment (siddhijain2) [reply contrib/babelfishpg_tsql/sql/information_schema_tsql.sql]: Addressed

File: contrib/babelfishpg_tsql/sql/information_schema_tsql.sql (hunk 2 above)
Comment (siddhijain2) [reply contrib/babelfishpg_tsql/sql/information_schema_tsql.sql]: Addressed

File: test/JDBC/input/key_column_usage-vu-prepare.sql (hunk 3 above)
Comment (siddhijain2) [reply test/JDBC/input/key_column_usage-vu-prepare.sql:7]: Addressed

File: test/JDBC/expected/key_column_usage-vu-verify.out [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+key_column_usage_vu_prepare_db#!#key_column_usage_vu_prepare_sc1#!#key_column_usage_vu_prepare_tbl4_pkey#!#key_column_usage_vu_prepare_db#!#key_column_usage_vu_prepare_sc1#!#key_column_usage_vu_prepare_tbl4#!#arg7#!#1
+key_column_usage_vu_prepare_db#!#dbo#!#key_column_usage_vu_prepare_tbl5_arg10_arg11_fkey#!#key_column_usage_vu_prepare_db#!#dbo#!#key_column_usage_vu_prepare_tbl5#!#arg10#!#1
+key_column_usage_vu_prepare_db#!#dbo#!#key_column_usage_vu_prepare_tbl5_arg10_arg11_fkey#!#key_column_usage_vu_prepare_db#!#dbo#!#key_column_usage_vu_prepare_tbl5#!#arg11#!#2
+key_column_usage_vu_prepare_db#!#dbo#!#key_column_usage_vu_prepare_tbl6_arg13_key#!#key_column_usage_vu_prepare_db#!#dbo#!#key_column_usage_vu_prepare_tbl6#!#arg13#!#1
+key_column_usage_vu_prepare_db#!#dbo#!#key_column_usage_vu_prepare_tbl6_arg15_fkey#!#key_column_usage_vu_prepare_db#!#dbo#!#key_column_usage_vu_prepare_tbl6#!#arg15#!#1
+key_column_usage_vu_prepare_db#!#dbo#!#key_column_usage_vu_prepare_tbl6_pkey#!#key_column_usage_vu_prepare_db#!#dbo#!#key_column_usage_vu_prepare_tbl6#!#arg12#!#1
```
Comment (Deepesh125) [inline test/JDBC/expected/key_column_usage-vu-verify.out:57]: Why column arg14 is not being returned here?

**Review by Deepesh125 — APPROVED** [review]

**Pull request merged**